pkg/util/vsphere/watcher/watcher.go

```diff
@@ -0,0 +1,656 @@
+// watch.
+func DefaultWatchedPropertyPaths() []string {
+	return []string{
+		"config.extraConfig",
+		"guest.ipStack",
+		"guest.net",
+		"summary.config.name",
+		"summary.guest",
+		"summary.overallStatus",
+		"summary.runtime.host",
+		"summary.runtime.powerState",
+		"summary.storage.timestamp",
+	}
+}
+
+const extraConfigNamespacedNameKey = "vmservice.namespacedName"
+
+// defaultIgnoredExtraConfigKeys returns the default set of extra config keys to
+// ignore.
+var defaultIgnoredExtraConfigKeys = []string{
+	backupapi.AdditionalResourcesYAMLExtraConfigKey,
+	backupapi.BackupVersionExtraConfigKey,
+	backupapi.ClassicDiskDataExtraConfigKey,
+	backupapi.DisableAutoRegistrationExtraConfigKey,
+	backupapi.EnableAutoRegistrationExtraConfigKey,
+	backupapi.PVCDiskDataExtraConfigKey,
+	backupapi.VMResourceYAMLExtraConfigKey,
+	"govcsim",
+	"guestinfo.metadata",
+	"guestinfo.metadata.encoding",
+	"guestinfo.userdata",
+	"guestinfo.userdata.encoding",
+	"guestinfo.vendordata",
+	"guestinfo.vendordata.encoding",
+	extraConfigNamespacedNameKey,
+}
+
+type moRef = vimtypes.ManagedObjectReference
+
+type lookupNamespacedNameFn func(context.Context, moRef) (string, string, bool)
+
+type Result struct {
+	// Namespace is the namespace to which the VirtualMachine resource belongs.
+	Namespace string
+
+	// Name is the name of the VirtualMachine resource.
+	Name string
+
+	// Ref is the ManagedObjectReference for the VM in vSphere.
+	Ref moRef
+
+	// Verified is true if the VirtualMachine resource identified by Namespace
+	// and Name has already been verified to exist in this Kubernetes cluster.
+	Verified bool
+}
+
+type Watcher struct {
+	mu sync.Mutex
+
+	cancel func()
+
+	client *vim25.Client
+
+	vm *view.Manager
+	lv *view.ListView
+	cv map[moRef]*view.ContainerView
+
+	// cvr is used to count the number of times a container has been added to
+	// the list view. If the Remove function is called on a container with a ref
+	// count of one, then the container will be removed from the list view and
+	// destroyed.
+	cvr map[moRef]map[string]struct{}
+
+	pc *property.Collector
+	pf *property.Filter
+
+	ignoredExtraConfigKeys map[string]struct{}
+	lookupNamespacedName   lookupNamespacedNameFn
+
+	err   error
+	errMu sync.RWMutex
+
+	chanDone   chan struct{}
+	chanResult chan Result
+}
+
+// Done returns a channel that is closed when the watcher is shutdown.
+func (w *Watcher) Done() <-chan struct{} {
+	return w.chanDone
+}
+
+// Result returns a channel on which new results are received.
+func (w *Watcher) Result() <-chan Result {
+	return w.chanResult
+}
+
+// Err returns the error that caused the watcher to stop.
+func (w *Watcher) Err() error {
+	w.errMu.RLock()
+	err := w.err
+	w.errMu.RUnlock()
+	return err
+}
+
+func (w *Watcher) setErr(err error) {
+	w.errMu.Lock()
+	w.err = err
+	w.errMu.Unlock()
+}
+
+func newWatcher(
+	ctx context.Context,
+	client *vim25.Client,
+	watchedPropertyPaths []string,
+	additionalIgnoredExtraConfigKeys []string,
```

Comment:
I see that we have this parameter to augment the list of ignored extraconfig keys. However, when we instantiate this watcher we are passing nil to it.

Suggestion: Does it make sense to define a additional source (like a predefined named configmap or secret or something else as well) of this list that can be set in a customer env if need be that could on the fly augment this? Case I am thinking about is say the user has some third party tool or their own orchestration that uses extraConfig keys as well, in such cases they might want to skip those  as well. (although technically that may very well lead to a no-op reconcile, but just waste a reconcile cycle that could have been avoided). 
I am almost certain you would have considered this case already :) and if so, was there a specific reasoning of not to do that? 

Also, I just literally looked at this MR so if you have covered this somewhere else, please feel free to point me to that.

Reply:
Allowing an external source of keys to ignore makes a ton of sense. Could you please file an issue for this idea? Thanks!